service/history/handler.go

```diff
@@ -124,11 +126,21 @@ func NewHandler(
 
 // Start starts the handler
 func (h *Handler) Start() {
+	if !atomic.CompareAndSwapInt32(
+		&h.status,
+		common.DaemonStatusInitialized,
+		common.DaemonStatusStarted,
+	) {
+		return
+	}
+
 	if h.GetClusterMetadata().IsGlobalNamespaceEnabled() {
-		var err error
-		h.publisher, err = h.GetMessagingClient().NewProducerWithClusterName(h.GetClusterMetadata().GetCurrentClusterName())
-		if err != nil {
-			h.GetLogger().Fatal("Creating kafka producer failed", tag.Error(err))
+		if h.GetClusterMetadata().GetReplicationConsumerConfig().Type == serviceConfig.ReplicationConsumerTypeKafka {
+			var err error
+			h.publisher, err = h.GetMessagingClient().NewProducerWithClusterName(h.GetClusterMetadata().GetCurrentClusterName())
+			if err != nil {
+				h.GetLogger().Fatal("Creating kafka producer failed", tag.Error(err))
```

Comment:
I guess it does not matter anymore as we are getting rid of Kafka, but in general preventing history service to come up if Kafka is down is a bad design choice.

Reply:
yep